# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserBridgeParent.cpp
# Commit: d6f212c67002
# Full Hash: d6f212c670027f013f8453181d9b1cfc9b4816dd
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-03-31 21:54:44
# Regressor Bug: 1699721
# File Overlap Count: 1
# Description:
#   Bug 1699721 - Part 2: Track BrowserParent lifecycles during process switches, r=kmag
#   
#   This patch contains a large number of changes around the process switching
#   mechanism in order to avoid issues which are caused by a mismatched
#   understanding of the state of the process switch between processes in the
# ==============================================================================

diff -r ef06d9764cf1 -r d6f212c67002 dom/ipc/BrowserBridgeParent.cpp
--- a/dom/ipc/BrowserBridgeParent.cpp	Wed Mar 31 15:37:48 2021 +0000
+++ b/dom/ipc/BrowserBridgeParent.cpp	Wed Mar 31 15:37:49 2021 +0000
@@ -41,6 +41,10 @@
     return NS_ERROR_UNEXPECTED;
   }
 
+  MOZ_DIAGNOSTIC_ASSERT(
+      !browsingContext->GetBrowserParent(),
+      "BrowsingContext must have had previous BrowserParent cleared");
+
   // Unfortunately, due to the current racy destruction of BrowsingContext
   // instances when Fission is enabled, while `browsingContext` may not be
   // discarded, an ancestor might be.
@@ -108,6 +112,8 @@
   mBrowserParent->SetOwnerElement(aParentBrowser->GetOwnerElement());
   mBrowserParent->InitRendering();
 
+  GetBrowsingContext()->SetCurrentBrowserParent(mBrowserParent);
+
   windowParent->Init();
   return NS_OK;
 }